src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/convert_bf16_fp32_fp16.cpp

```diff
@@ -0,0 +1,146 @@
+        const auto weights_left = init_compressed_weights_subgraph(ov::element::bf16, 2);
+        const auto weights_right = init_compressed_weights_subgraph(ov::element::bf16, 3);
+        auto matmul_res = std::make_shared<ov::op::v1::Add>(weights_left, weights_right);
+        return std::make_shared<ov::Model>(ov::NodeVector{matmul_res}, ov::ParameterVector{}, "BF16WeightsDecompression");
+    }
+};
+
+/*
+TEST_F(BF16WeightsDecompression, TypeCheck) {
+    std::shared_ptr<ov::Model> subgraph = init_subgraph();
+    ov::CompiledModel model = core->compile_model(subgraph, targetDevice, {ov::hint::inference_precision(ov::element::f32)});
+    bool convert_found = false;
+    std::cout << "entering " << model.get_runtime_model()->get_ops().size() << std::endl;
+    for ( const auto& n : model.get_runtime_model()->get_ordered_ops() ) {
+        std::cout << "znaleziono " << n->get_friendly_name() << std::endl;
+        if ( n->get_friendly_name() == "conversion_bf16_to_f32" ) {
+            ASSERT_EQ(n->get_input_element_type(0),  ov::element::bf16);
+            ASSERT_EQ(n->get_output_element_type(0),  ov::element::f32);
+            convert_found = true;
+        }
+    }
+    ASSERT_TRUE(convert_found);
+}
+*/
+TEST_F(BF16WeightsDecompression, Inference_input_3) {
+    std::shared_ptr<ov::Model> function = init_subgraph();
+    ov::CompiledModel cM = core->compile_model(function, targetDevice, {ov::hint::inference_precision(ov::element::f32)});
+    auto request = cM.create_infer_request();
+    request.infer();
+    auto out_tensor = request.get_output_tensor();
+    for ( size_t i = 0 ; i < out_tensor.get_size(); i++ ) {
+        ASSERT_FLOAT_EQ(reinterpret_cast<float*>(out_tensor.data())[i], 12);
+    }
+}
+
+TEST_F(BF16WeightsDecompression, Inference_input_3_f16) {
+    std::shared_ptr<ov::Model> function = init_subgraph();
+    ov::CompiledModel cM = core->compile_model(function, targetDevice, {ov::hint::inference_precision(ov::element::f16)});
+    auto request = cM.create_infer_request();
+    request.infer();
+    auto out_tensor = request.get_output_tensor();
+    for ( size_t i = 0 ; i < out_tensor.get_size(); i++ ) {
+        ASSERT_FLOAT_EQ(reinterpret_cast<float*>(out_tensor.data())[i], 12);
+    }
+}
+
+TEST_P(BF16WeightsDecompression, Inference_without_convert) {
+    float given;
+    float expected;
+    std::tie(given, expected) = GetParam();
+    std::shared_ptr<ov::Model> function_without_convert = init_subgraph_without_convert(given);
+    ov::CompiledModel model_without_convert = core->compile_model(function_without_convert, targetDevice, {ov::hint::inference_precision(ov::element::f32)});
+    auto request = model_without_convert.create_infer_request();
+    request.infer();
+    auto out_tensor = request.get_output_tensor();
+    for ( size_t i = 0 ; i < out_tensor.get_size(); i++ ) {
+        ASSERT_NEAR(reinterpret_cast<unsigned short*>(out_tensor.data())[i], expected, 0.2f); //3072
```

Comment:
it is difficult to understand the meaning of this comment. `3072` could you explain more?

Reply:
sorry it is here by mistake, I will delete it